pyface/data_view/value_types/enum_value.py

```diff
@@ -0,0 +1,162 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from traits.api import Callable, List
+
+from .editable_value import EditableValue
+
+
+class EnumValue(EditableValue):
+    """ Editable value that takes one of a collection of pre-set values.
+
+    Each value can be associated with text, colors and images by supplying
+    functions ``format``, ``colors`` and ``images``, respectively.
+    """
+
+    #: The list of values which are allowed for the value.
+    values = List()
+
+    #: A function that converts a value to a string for display.
+    format = Callable(str, update_value_type=True)
+
+    #: A map from valid values to colors.
+    colors = Callable(None, update_value_type=True)
+
+    #: A map from valid values to images.
+    images = Callable(None, update_value_type=True)
+
+    def is_valid(self, model, row, column, value):
```

Comment:
No.  Data flow still goes through here.  For example, if someone decides to use a text editor instead of a combo box, we still need this validation.  Data flow in the `QStyleItemDelegate` that I have sketched out is still through `setData` irrespective of the item editor, which will call this code.